### PR TITLE
Keeping critical files/Pending Reboot

### PR DESCRIPTION
**Pull request by KnifMelti**
See https://github.com/Romanitho/Winget-AutoUpdate/discussions/135#discussion-4262191

### CODE REVIEW

File: Winget-AutoUpdate-Install.ps1 [hunk 1]
```diff
@@ -176,13 +183,19 @@ function Install-WingetAutoUpdate {
             New-Item -ItemType Directory -Force -Path $WingetUpdatePath | Out-Null
         }
         else {
-            Remove-Item -Path "$WingetUpdatePath\*" -Exclude *.log -Recurse -Force
+            if (!$NoClean) {
+                Remove-Item "$WingetUpdatePath\*" -Force -Recurse -Exclude "*.log"
+            }
+            else {
+                #Keep critical files
+                Get-ChildItem -Path $WingetUpdatePath -Exclude included_apps.txt,mods,logs | Remove-Item -Recurse -Force
```
Comment (Romanitho) [inline Winget-AutoUpdate-Install.ps1:191]: we should also keep the excluded_apps.txt no ?

Comment (KnifMelti) [reply Winget-AutoUpdate-Install.ps1]: Yup, that was for a loop of computers.
Think I'm done.

**Comment by KnifMelti** [comment]
Now done (I think)!

**Comment by KnifMelti** [comment]
Full circle.

**Comment by KnifMelti** [comment]
Pending reboot should not be tested until upgrade fails, then breaking the loop so no install is tried for that failed app.
Let me come up with something else, please.
Hold your review.

**Review by Romanitho — COMMENTED** [review]
Seems good :)
Thanks for your work

**Comment by KnifMelti** [comment]
Let's hope..  .. I've tested a lot...
... even so, could have missed something...
Thank you for WAU!

**Review by KnifMelti — COMMENTED** [review]
All good

**Pull request merged**